### ISSUE TITLE
Style/RedundantSelf creates false positive with multiple assignment

### ISSUE THREAD

**Issue by dmitryv** [rubocop-hq/rubocop#7353]
```
 phone, area_code = parse_phone(self.phone)
                                ^^^^^^^^^^ Redundant `self` detected.

```
--------

## Expected behavior

Style/RedundantSelf Cop should leave this self-ish line of code intact

## Actual behavior

RuboCop autocorrects this line by removing the "self." part

## Steps to reproduce the problem

Failing unit test to reproduce the issue here:

https://github.com/dmitryv/rubocop/commit/bbc683c1a4d9c59050ebbee9490745b918ba1d91

Or another example:

```
:001 > class A
:002?>     attr_reader :x
:003?>
:004 >       def initialize
:005?>         @x = 1
:006?>       end
:007?>
:008 >       def phone
:009?>         x = self.x
:010?>         x
:011?>       end
:012?>   end
 => :phone
:013 > A.new.phone
 => 1
```

```
:014 > class A
:015?>     attr_reader :x
:016?>
:017 >       def initialize
:018?>         @x = 1
:019?>       end
:020?>
:021 >       def phone
:022?>         x = x
:023?>         x
:024?>       end
:025?>   end
 => :phone
:026 > A.new.phone
 => nil
```

## RuboCop version

0.74.0 (using Parser 2.6.4.0, running on ruby 2.3.3 x86_64-darwin18)

**Comment by koic** [comment]
Thanks for the feedback. This issue has two false positive cases.

First: When multiple assignment ... `a, b = self.a`
Second: When method argument ... `a = do_something(self.a)`

First, I opened a PR #7354 which solves the first false positive case.

**Comment by koic** [comment]
I reopen this issue and I opened a PR #7356 which solves the second false positive case.
